charms/istio-pilot/src/resources_handler.py

```diff
@@ -35,30 +35,34 @@ def __init__(self, app_name, model_name):
         self.env = Environment(loader=FileSystemLoader('src'))
 
     def delete_resource(
-        self, obj, namespace=None, ignore_not_found=False, ignore_unauthorized=False
+        self,
+        obj,
+        namespace=None,
+        ignore_not_found=False,
     ):
         try:
             self.lightkube_client.delete(type(obj), obj.metadata.name, namespace=namespace)
-        except ApiError as err:
-            self.log.exception("ApiError encountered while attempting to delete resource.")
-            if err.status.message is not None:
-                if "not found" in err.status.message and ignore_not_found:
-                    self.log.error(f"Ignoring not found error:\n{err.status.message}")
-                elif "(Unauthorized)" in err.status.message and ignore_unauthorized:
-                    # Ignore error from https://bugs.launchpad.net/juju/+bug/1941655
-                    self.log.error(f"Ignoring unauthorized error:\n{err.status.message}")
-                else:
-                    self.log.error(err.status.message)
-                    raise
-            else:
-                raise
+
+        # For some reason, on 404 Not Found errors the raised exception
+        # is httpx.HTTPStatusError instead of an ApiError
+        # FIXME: find out why this is the case, probably something missing
+        # in upstream lightkube
+        except httpx.HTTPStatusError as err:
+            self.log.warning(
+                "HTTPStatusError encountered while attempting to delete resource.\n"
+                "If ignore_not_found is set to True, this WARNING can be dismissed."
+            )
+            if err.response.status_code == 404 and ignore_not_found:
+                self.log.warning(f"Ignoring not found error for: {obj.kind}/{obj.metadata.name}")
+                return
```

Comment:
Not specifically something you're changing, but should this be a `return` here?  It feels more like it should be one of:
* `continue`: If this is a warning and might happen due to a bad object in a yaml, but it is ignorable and we should keep going
* `raise`: If this a real error and things should stop right away